asmcomp/cmm.ml

```diff
@@ -32,6 +32,39 @@ let size_component = function
   | Int -> Arch.size_int
   | Float -> Arch.size_float
 
+let lub_component comp1 comp2 =
+  match comp1, comp2 with
+  | Int, Int -> Int
+  | Int, Addr -> Addr
+  | Int, Val -> Val
+  | Addr, Int -> Addr
+  | Addr, Addr -> Addr
+  | Addr, Val -> Val
+  | Val, Int -> Val
+  | Val, Addr -> Val
+  | Val, Val -> Val
+  | Float, Float -> Float
+  | (Int | Addr | Val), Float
+  | Float, (Int | Addr | Val) ->
+    (* Float unboxing code must be sure to avoid this case. *)
+    assert false
+
+let ge_component comp1 comp2 =
```

Comment:
Couldn't this be implemented as `comp1 = lub_component comp1 comp2`? That would avoid having to check the code twice